core/templates/dev/head/profile/Signup.js

```diff
@@ -67,7 +75,7 @@ oppia.controller('Signup', [
         username: $scope.username
       }).success(function(data) {
         if (data.username_is_taken) {
-          $scope.warningText = 'Sorry, this username is already taken.'
+          $scope.warningText = 'I18N_SIGNUP_ERROR_USERNAME_TAKEN'
```

Comment:
I think the same, those string are ugly. But the problem of making the translation in the js is the html will contain the translation message and not the translation id. This means, when you click the footer to change to Spanish, the message does no change until you pass again through the js function.
I left it untouched for now, another solution can be a map between error types and error strings (and maybe the conditions that trigger each error) as a global variable.